### PR TITLE
feat(timeline): implement timeline v2 - INNO-1451

### PR DESCRIPTION
**Pull request by emeryro**
# PR description

Add [new version of the timeline](https://webgate.ec.europa.eu/CITnet/confluence/pages/viewpage.action?pageId=861357293)

To test the behavior when collapsing the timeline, a button is available in storybook to add dummy content below the timeline.

To check retro compatibility, import Timeline instead of Timeline2 in story:
`import Timeline from '../src/Timeline2';`

### CODE REVIEW

**Comment by HAL-Patch-INNO** [comment]
Deploy preview for *europa-component-library* ready!

Built with commit ecb6eb3140b9a55ef56e35607f3262512dca2d14

https://deploy-preview-1124--europa-component-library.netlify.com

**Comment by HAL-Patch-INNO** [comment]
Deploy preview for *europa-component-library* ready!

Built with commit 6f2a66dd97126c617338100498eed1887bcaca37

https://deploy-preview-1124--europa-component-library.netlify.com

File: src/systems/ec/implementations/react/components/timeline/package.json [hunk 1]
```diff
@@ -12,7 +12,9 @@
     "@ecl/ec-react-component-button": "^2.4.0"
   },
   "devDependencies": {
-    "@ecl/ec-specs-timeline": "^2.4.0"
+    "@ecl/ec-specs-timeline": "^2.4.0",
+    "html-react-parser": "^0.7.0",
```
Comment (yhuard) [inline src/systems/ec/implementations/react/components/timeline/package.json]:
```suggestion
    "html-react-parser": "0.7.0",
```

File: src/systems/ec/implementations/react/components/timeline/package.json [hunk 2]
```diff
@@ -12,7 +12,9 @@
...
+    "lorem-ipsum": "^2.0.2"
```
Comment (yhuard) [inline src/systems/ec/implementations/react/components/timeline/package.json]:
```suggestion
    "lorem-ipsum": "2.0.2"
```

File: src/systems/ec/implementations/react/components/timeline/src/Timeline2.jsx [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+  toggleCollapsed,
+  toggleExpanded,
+  button,
+  className,
+  ...props
+}) => {
+  const classNames = classnames(className, 'ecl-timeline2');
+
+  // Calculate collapse threshold
+  const threshold =
```
Comment (yhuard) [inline src/systems/ec/implementations/react/components/timeline/src/Timeline2.jsx:21]: is this logic set in stone (i.e. always THE logic to follow)? should it be part of the component or should we create a helper for it?

File: src/systems/ec/implementations/react/components/timeline/src/Timeline2.jsx [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+    items.length > 12
+      ? Math.min(Math.floor(items.length * 0.2), 7)
+      : items.length;
+  const hidenItemsLength = items.length - threshold - 2;
+
+  const toggleLabelCollapsed = toggleCollapsed
```
Comment (yhuard) [inline src/systems/ec/implementations/react/components/timeline/src/Timeline2.jsx:27]: wondering if this logic is necessary or if it's up to the developer/consumer to implement it

when the data is available, the developer "knows" how many items there are, he could update the label on his side before feeding the component

File: src/systems/ec/implementations/vanilla/packages/ec-component-timeline/ec-component-timeline.js [hunk 5]
```diff
@@ -4,7 +4,10 @@ export class Timeline {
   constructor(
     element,
     {
-      buttonSelector: buttonSelector = '[data-ecl-timeline-button]',
+      toggleSelector: toggleSelector = '[data-ecl-timeline-button]',
```
Comment (yhuard) [inline src/systems/ec/implementations/vanilla/packages/ec-component-timeline/ec-component-timeline.js]: ⚠️  renaming the param = breaking change

File: src/systems/eu/implementations/react/components/timeline/package.json [hunk 6]
```diff
@@ -12,7 +12,9 @@
     "@ecl/eu-react-component-button": "^2.4.0"
   },
   "devDependencies": {
-    "@ecl/eu-specs-timeline": "^2.4.0"
+    "@ecl/eu-specs-timeline": "^2.4.0",
+    "html-react-parser": "^0.7.0",
```
Comment (yhuard) [inline src/systems/eu/implementations/react/components/timeline/package.json]:
```suggestion
    "html-react-parser": "0.7.0",
```

File: src/systems/eu/implementations/react/components/timeline/package.json [hunk 7]
```diff
@@ -12,7 +12,9 @@
...
+    "lorem-ipsum": "^2.0.2"
```
Comment (yhuard) [inline src/systems/eu/implementations/react/components/timeline/package.json]:
```suggestion
    "lorem-ipsum": "2.0.2"
```

**Review by yhuard — CHANGES_REQUESTED** [review]
I haven't checked the design in details but it looks nice! 👍 Now:
- can you add, in the existing example or in a new example, an item with a title please?
- can you make sure it is resilient? (doesn't change when you activate "test-fake-global-rules" CSS global rules)

File: src/systems/ec/implementations/react/components/timeline/src/Timeline2.jsx (hunk 3 above)
Comment (emeryro) [reply src/systems/ec/implementations/react/components/timeline/src/Timeline2.jsx:21]: it is written in the specs
image

File: src/systems/ec/implementations/react/components/timeline/src/Timeline2.jsx (hunk 4 above)
Comment (emeryro) [reply src/systems/ec/implementations/react/components/timeline/src/Timeline2.jsx:27]: well, he knows how many items there are in total, yes. But he may not be aware of the calculus to get the collapse/expand thershold. So he doesn't know how many are hidden

File: src/systems/ec/implementations/react/components/timeline/src/Timeline2.jsx (hunk 3 above)
Comment (yhuard) [reply src/systems/ec/implementations/react/components/timeline/src/Timeline2.jsx:21]: I know it is written in the specs, but this logic doesn't seem "perfect". It already starts with "more than 13-14 items", it's not precise enough. "should make use". "if the show/hide is used". To me, it's more a guideline than a rule.

File: src/systems/ec/implementations/react/components/timeline/src/Timeline2.jsx (hunk 4 above)
Comment (yhuard) [reply src/systems/ec/implementations/react/components/timeline/src/Timeline2.jsx:27]: I understand your point of view. In the end, it depends on whether or not the developer has to implement the calculus or not. Let's close this discussion and continue on the one above ⬆️

**Review by yhuard — APPROVED** [review]

**Pull request merged**